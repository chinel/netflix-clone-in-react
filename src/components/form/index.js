import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
