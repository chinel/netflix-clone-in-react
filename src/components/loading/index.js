import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
