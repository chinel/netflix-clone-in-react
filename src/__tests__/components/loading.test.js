import React from "react";
import { render } from "@testing-library/react";
import { Loading } from "../../components";

describe("<Loading />", () => {
  it("renders the <Loading /> component", () => {
    const { container, getByTestId } = render(
      <Loading src="/images/karl.png" data-testid="loading" />
    );
  });
});
