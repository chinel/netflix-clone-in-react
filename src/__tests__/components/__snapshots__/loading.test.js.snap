// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders the <Loading /> component 1`] = `
<div
  class="loading__Spinner-sc-1q1z54u-0 eTeIrC"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-1q1z54u-1 iMWTOT"
    data-testid="loading-picture"
    src="/images/users//images/karl.png.png"
  />
</div>
`;

exports[`<Loading /> renders the <Loading.ReleaseBody/> 1`] = `null`;
